### ISSUE TITLE
Error: couldn't generate KfApp:  (kubeflow.error): Code 500 with message: kfApp Generate failed for kustomize: kustomize generate failed Error:  (kubeflow.error): Code 500 with message: couldn't copy application istio-ingress

### ISSUE THREAD

**Issue by 14sudharsan** [kubeflow/kubeflow#4199]
when i tried to install kubeflow v0.6 in EKS cluster with help of this link 
https://master.kubeflow.org/docs/aws/deploy/install-kubeflow

root@ip-172-31-31-235:~/hcl-cluster-kf# **kubectl version**
Client Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.10-eks-f12431", GitCommit:"f12431e9d92fa990fbbc2b52006351d381bd4bed", GitTreeState:"clean", BuildDate:"2019-08-01T23:43:58Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.10-eks-5ac0f1", GitCommit:"5ac0f1d9ab2c254ea2b0ce3534fd72932094c6e1", GitTreeState:"clean", BuildDate:"2019-08-20T22:39:46Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}

root@ip-172-31-31-235:~/hcl-cluster-kf# **eksctl version**
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.5.3"}

**Getting below Error**
root@ip-172-xx-xx-xxx:~/testing-cluster-kf# kfctl generate all -V
INFO[0000] Downloading /root/testing-cluster-kf/app.yaml to /tmp/790876065/app.yaml  filename="v1alpha1/application_types.go:334"
INFO[0000] Writing stripped KfDef to /root/testing-cluster-kf/app.yaml  filename="v1alpha1/application_types.go:626"
INFO[0000] Downloading /root/testing-cluster-kf/app.yaml to /tmp/584471180/app.yaml  filename="v1alpha1/application_types.go:334"
INFO[0000] Caller ARN Info: {
  Account: "245980173641",
  Arn: "arn:aws:iam::245980173641:user/rapydipa",
  UserId: "AIDATSRMPSVEVZLSCJBC6"
}  filename="utils/awsutil.go:38"
INFO[0000] AWS infrastructure configs already exist in directory /root/testing-cluster-kf/aws_config  filename="aws/aws.go:323"
INFO[0000] Reading config file: /root/testing-cluster-kf/aws_config/cluster_features.yaml  filename="aws/aws.go:768"
INFO[0000] Reading config file: /root/testing-cluster-kf/aws_config/cluster_features.yaml  filename="aws/aws.go:768"
INFO[0000] Writing stripped KfDef to /root/testing-cluster-kf/app.yaml  filename="v1alpha1/application_types.go:626"
INFO[0000] Initializing kustomize package map            filename="kustomize/kustomize.go:150"
INFO[0000] Component path map: {}                        filename="kustomize/kustomize.go:171"
INFO[0000] Writing stripped KfDef to /root/testing-cluster-kf/app.yaml  filename="v1alpha1/application_types.go:626"
INFO[0000] Processing application: istio-crds            filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: istio-install         filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: istio                 filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: application-crds      filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: application           filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: metacontroller        filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: argo                  filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: centraldashboard      filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: webhook               filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: bootstrap             filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: jupyter-web-app       filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: katib-db              filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: katib-manager         filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: katib-controller      filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: katib-ui              filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: metrics-collector     filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: metadata              filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: suggestion            filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: notebook-controller   filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: pytorch-job-crds      filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: pytorch-operator      filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: spartakus             filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: tensorboard           filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: tf-job-operator       filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: api-service           filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: minio                 filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: mysql                 filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: persistent-agent      filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: pipelines-runner      filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: pipelines-ui          filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: pipelines-viewer      filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: scheduledworkflow     filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: profiles              filename="kustomize/kustomize.go:606"
INFO[0000] Processing application: istio-ingress         filename="kustomize/kustomize.go:606"
Error: couldn't generate KfApp:  (kubeflow.error): Code 500 with message: kfApp Generate failed for kustomize: kustomize generate failed Error:  (kubeflow.error): Code 500 with message: couldn't copy application istio-ingress
Usage:
  kfctl generate [all(=default)|k8s|platform] [flags]

Flags:
      --email string      email if '--platform gcp'
  -h, --help              help for generate
      --hostname string   hostname if '--platform gcp'
      --ipName string     ipName if '--platform gcp'
      --mount-local       mount-local if '--platform minikube'
  -V, --verbose           verbose output default is false
      --zone string       zone if '--platform gcp'

couldn't generate KfApp:  (kubeflow.error): Code 500 with message: kfApp Generate failed for kustomize: kustomize generate failed Error:  (kubeflow.error): Code 500 with message: couldn't copy application istio-ingress

**Comment by mikejmills** [comment]
I have a similar issue 

`Error: couldn't apply KfApp:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: couldn't create resources from istio-install Error: no matches for kind "Deployment" in version "extensions/v1beta1"
Usage:
  kfctl apply [all(=default)|k8s|platform] [flags]

Flags:
  -h, --help      help for apply
  -V, --verbose   verbose output default is false

couldn't apply KfApp:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: couldn't create resources from istio-install Error: no matches for kind "Deployment" in version "extensions/v1beta1"`

**Comment by kevincasadodsc** [comment]
I believe this is the same as https://github.com/kubeflow/kubeflow/issues/4057. Looks like it's in pr here https://github.com/kubeflow/manifests/pull/360 but has yet to be merged.

**Comment by MostafaGazar** [comment]
A quick workaround if you are using [multipass](http://multipass.run/) with [microk8s](http://microk8s.io/) is to install v1.14. I just tried it and it worked fine. 

Follow the steps in https://tutorials.ubuntu.com/tutorial/get-started-kubeflow but install microk8s via `sudo snap install microk8s --classic --channel=1.14/stable` instead.

**Comment by mikejmills** [comment]
Thanks for the suggestion but no luck same error.

**Comment by Jeffwan** [comment]
Sorry guys I will try to resolve this issue and make aws components merge into v0.6-branch

**Comment by samuelthan** [comment]
> I believe this is the same as #4057. Looks like it's in pr here [kubeflow/manifests#360](https://github.com/kubeflow/manifests/pull/360) but has yet to be merged.

Yeah, i tried to use the work around by using master as the branch for the manifest.  But the work around no longer fix the issue.

**Comment by MostafaGazar** [comment]
I used the workaround with tag v0.6.2

**Comment by William8Work** [comment]
Similar issue here. 

Error: couldn't generate KfApp:  (kubeflow.error): Code 500 with message: kfApp Generate failed for kustomize: kustomize generate failed Error:  (kubeflow.error): Code 500 with message: couldn't copy application istio-ingress
Usage:
  kfctl generate [all(=default)|k8s|platform] [flags]

Flags:
      --email string      email if '--platform gcp'
  -h, --help              help for generate
      --hostname string   hostname if '--platform gcp'
      --ipName string     ipName if '--platform gcp'
      --mount-local       mount-local if '--platform minikube'
  -V, --verbose           verbose output default is false
      --zone string       zone if '--platform gcp'

**Comment by zmdelima** [comment]
Hi guys, we are encountering this issue too (using v0.6.2 kfctl_aws.yaml as it points to v0.6-branch repo for kubeflow/kubeflow and kubeflow/manisfests)
So we have been experimenting with the commits (for days) and we use these:

with `CONFIG` set to the path of kfctl_aws.yaml

For kubeflow/kubeflow:
```
sed -i "s@https://github.com/kubeflow/manifests/archive/7a373809a663ded140f9b70a3e879a39f6e88479.tar.gz@https://github.com/kubeflow/kubeflow/archive/9804feb9fc23fc30075632a857087f4b529294e2.tar.gz@" ${CONFIG}
```
For kubeflow/manifests:
```
sed -i "s@https://github.com/kubeflow/manifests/archive/v0.6-branch.tar.gz@https://github.com/kubeflow/manifests/archive/7a373809a663ded140f9b70a3e879a39f6e88479.tar.gz@" ${CONFIG}
```

I hope this helps.

**Comment by plaffitte** [comment]
@zmdelima This didn't solve the problem. 
When initializing with `kfctl init ${KFAPP} --config=${CONFIG} -V` I could see it fetched the right version, according to the commit you pointed to, but running `kfctl generate all -v` still returned the same error.

**Comment by zmdelima** [comment]
@plaffitte I am currently using v0.6.1/bootstrap instead of v0.6.2/bootstrap `kfctl_aws.yaml` source as the major difference from them besides the repository targets is having an `- application` for the kustomize of `path: tf-training/tf-job-operator` as this works for me.

These are my step by step cli commands:
```
export CONFIG=~/kfctl_aws.yaml

curl -Lo ${CONFIG} https://raw.githubusercontent.com/kubeflow/kubeflow/v0.6.1/bootstrap/config/kfctl_aws.yaml

# We create an anonymous kubeflow namespace
kubectl create namespace kubeflow-anonymous

# Replace lines for the InstanceRole name and AWS Region
sed -i "s@eksctl-kubeflow-aws-nodegroup-ng-a2-NodeInstanceRole-xxxxxxx@$ROLE_NAME@" ${CONFIG}
sed -i "s@us-west-2@$AWS_REGION@" ${CONFIG}

# Replace repository targets
sed -i "s@https://github.com/kubeflow/kubeflow/archive/master.tar.gz@https://github.com/kubeflow/kubeflow/archive/9804feb9fc23fc30075632a857087f4b529294e2.tar.gz@" ${CONFIG}

sed -i "s@https://github.com/kubeflow/manifests/archive/master.tar.gz@https://github.com/kubeflow/manifests/archive/7a373809a663ded140f9b70a3e879a39f6e88479.tar.gz@" ${CONFIG}
```

Reference to changes on v0.6 kfctl_aws.yaml file:
https://github.com/kubeflow/kubeflow/commits/v0.6-branch/bootstrap/config/kfctl_aws.yaml

Note: If there are repercussions on the missing `application overlay` to `tf-training` please inform, thank you.

**Comment by zmdelima** [comment]
@karlschriek your error is not related on the deployment of the kubeflow in the EKS Cluster but more on the resolution of your pods.

1. Please check all your pods via `kubectl get pods --all-namespaces`
2. `kubectl delete pod/<pod-name> -n <pod's namespace>` for every crashing pod

if `kubectl get ingress -n istio-system` returns `no resources found` please check your EKS VPC
[eksctl VPC docs here](https://eksctl.io/usage/vpc-networking/).

Good luck.

**Comment by ardeshir** [comment]
INFO[0000] Creating AWS infrastructure configs in directory /home/ec2-user/environment/eksworkshop-eksctl/aws_config  filename="aws/aws.go:314"
Error: couldn't generate KfApp:  (kubeflow.error): Code 500 with message: coordinator Generate failed for aws:  (kubeflow.error): Code 400 with message: Could not generate cluster configs under aws_config Error: Could not copy /home/ec2-user/environment/eksworkshop-eksctl/.cache/kubeflow/kubeflow-master/deployment/aws/infra_configs/cluster_config.yaml to /home/ec2-user/environment/eksworkshop-eksctl/aws_config/cluster_config.yaml Error cannot create directory: open /home/ec2-user/environment/eksworkshop-eksctl/.cache/kubeflow/kubeflow-master/deployment/aws/infra_configs/cluster_config.yaml: no such file or directory
Usage:
  kfctl generate [all(=default)|k8s|platform] [flags]

Flags:
      --email string      email if '--platform gcp'
  -h, --help              help for generate
      --hostname string   hostname if '--platform gcp'
      --ipName string     ipName if '--platform gcp'
      --mount-local       mount-local if '--platform minikube'
  -V, --verbose           verbose output default is false
      --zone string       zone if '--platform gcp'

couldn't generate KfApp:  (kubeflow.error): Code 500 with message: coordinator Generate failed for aws:  (kubeflow.error): Code 400 with message: Could not generate cluster configs under aws_config Error: Could not copy /home/ec2-user/environment/eksworkshop-eksctl/.cache/kubeflow/kubeflow-master/deployment/aws/infra_configs/cluster_config.yaml to /home/ec2-user/environment/eksworkshop-eksctl/aws_config/cluster_config.yaml Error cannot create directory: open /home/ec2-user/environment/eksworkshop-eksctl/.cache/kubeflow/kubeflow-master/deployment/aws/infra_configs/cluster_config.yaml: no such file or directory

**Comment by ardeshir** [comment]
$ kfctl version
kfctl v0.6.1-rc.2-1-g3a37cbc6

**Comment by ardeshir** [comment]
$ kfctl version
kfctl v0.6.2-0-g47a0e4c7

**Comment by zmdelima** [comment]
@ardeshir I suggest you read my quoted comment below to @plaffitte's case.

As we are waiting for the resolution of the aws dependencies since the changes in the configuration directories for kfctl_aws.yaml

> @plaffitte I am currently using v0.6.1/bootstrap (as this works for me) instead of v0.6.2/bootstrap `kfctl_aws.yaml` source. The major difference from them besides the repository targets is having an `- application` for the kustomize of `path: tf-training/tf-job-operator`.
> 
> These are my step by step cli commands:
> 
> ```
> export CONFIG=~/kfctl_aws.yaml
> 
> curl -Lo ${CONFIG} https://raw.githubusercontent.com/kubeflow/kubeflow/v0.6.1/bootstrap/config/kfctl_aws.yaml
> 
> # We create an anonymous kubeflow namespace
> kubectl create namespace kubeflow-anonymous
> 
> # Replace lines for the InstanceRole name and AWS Region
> sed -i "s@eksctl-kubeflow-aws-nodegroup-ng-a2-NodeInstanceRole-xxxxxxx@$ROLE_NAME@" ${CONFIG}
> sed -i "s@us-west-2@$AWS_REGION@" ${CONFIG}
> 
> # Replace repository targets
> sed -i "s@https://github.com/kubeflow/kubeflow/archive/master.tar.gz@https://github.com/kubeflow/kubeflow/archive/9804feb9fc23fc30075632a857087f4b529294e2.tar.gz@" ${CONFIG}
> 
> sed -i "s@https://github.com/kubeflow/manifests/archive/master.tar.gz@https://github.com/kubeflow/manifests/archive/7a373809a663ded140f9b70a3e879a39f6e88479.tar.gz@" ${CONFIG}
> ```
> 
> Reference to changes on v0.6 kfctl_aws.yaml file:
> https://github.com/kubeflow/kubeflow/commits/v0.6-branch/bootstrap/config/kfctl_aws.yaml
> 
> Note: If there are repercussions on the missing `application overlay` to `tf-training` please inform, thank you.

**Comment by ardeshir** [comment]
Thanks @zmdelinma, I have followed that work around.

**Comment by ardeshir** [comment]
Thank you @zmdelima, using v0.6.2 and updated the targets as mentioned above got me past 
* $ kfctl generate all -V 
But kfctl apply all -V  fails   

INFO[0002] creating Service/istio-ingressgateway         filename="kustomize/kustomize.go:447"
Error: couldn't apply KfApp:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: couldn't create resources from istio-install Error: Service "istio-ingressgateway" is invalid: spec.ports[1].nodePort: Invalid value: 31380: provided port is already allocated
Usage:
  kfctl apply [all(=default)|k8s|platform] [flags]

Flags:
  -h, --help      help for apply
  -V, --verbose   verbose output default is false

couldn't apply KfApp:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: couldn't create resources from istio-install Error: Service "istio-ingressgateway" is invalid: spec.ports[1].nodePort: Invalid value: 31380: provided port is already allocated

**Comment by zmdelima** [comment]
@ardeshir that error means your istio-ingress pods are occupying the 31380 port, what I usually do is:

```
# go to your kubeflow app directory
cd ${KFAPP} 
# delete istio-system namespace
 kubectl delete namespace istio-system
# Generate the configurations again for istio-system
kfctl generate all -V
# Apply configurations once more
kfctl apply all -V
```

**Comment by ardeshir** [comment]
Thanks!  @zmdelima cleared istio-system and applied once more, and all the pods are running now. $ kubectl get pods -n kubeflow   shows them all running!

**Comment by dping1** [comment]
I was able to follow the workaround to install Kubeflow on EKS 1.14.  But the notebook-controller-deployment kep on crashing.  The log had the following error message in it

2019-10-13T22:20:03.059Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
2019-10-13T22:20:03.060Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "notebook", "source": "kind source: /, Kind="}
2019-10-13T22:20:03.060Z	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "Notebook.kubeflow.org", "error": "no matches for kind \"Notebook\" in version \"kubeflow.org/v1beta1\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/source/source.go:88
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Watch
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:122
sigs.k8s.io/controller-runtime/pkg/builder.(*Builder).doWatch
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/builder/controller.go:162
sigs.k8s.io/controller-runtime/pkg/builder.(*Builder).Build
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/builder/controller.go:151
github.com/kubeflow/kubeflow/components/notebook-controller/controllers.(*NotebookReconciler).SetupWithManager
	/workspace/controllers/notebook_controller.go:460
main.main
	/workspace/main.go:70
runtime.main
	/usr/local/go/src/runtime/proc.go:200
2019-10-13T22:20:03.060Z	ERROR	setup	unable to create controller	{"controller": "Notebook", "error": "no matches for kind \"Notebook\" in version \"kubeflow.org/v1beta1\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
main.main
	/workspace/main.go:71
runtime.main
	/usr/local/go/src/runtime/proc.go:200

**Comment by HarryCaveMan** [comment]
So I used the repo workaround and got istio copied and it failed again trying to coppy "katib-db"

**Comment by JohannesTK** [comment]
I'm getting no address for the ingress with the proposed solution:

```
$ kubectl get ingress -n istio-system
NAME            HOSTS   ADDRESS   PORTS   AGE
istio-ingress   *                 80      10m
```

Any solutions for that?

**Comment by shaunc** [comment]
@HarryCaveMan -- any workaround? I'm facing that issue as well. tried manually removing katib-related resources, but no luck yet.

**Comment by plaffitte** [comment]
@zmdelima Thanks for the tip but I still get the same error when running `kfctl generate all -v`...

**Comment by Jeffwan** [comment]
Sorry guys, I am kinda overwhelmed with other stuff recently and I will either
- give workaround on 0.6.2
- give 0.7 setups.
- Fix this issue with community folks

by this weekend.

**Comment by Jeffwan** [comment]
1. 0.6.2 branch, for people failed to general manifest, please change manifest branch from `master` to this. I use my own repo here instead.

```
    - name: manifests
      root: manifests-0.6-branch
      uri: https://github.com/jeffwan/manifests/archive/cherry-pick-aws-manifests.tar.gz
```

2. For user failed to create ALB, please `kubectl delete pod alb_pod_xxx -n kubeflow`. I am not sure why alb can not fetch pending istio here.. Need more investigation

3. I am working on 0.7. Doc will be released this week.  I will resolve all knowns issues in 0.7 release.

**Comment by Jeffwan** [comment]
Updates: All 0.7 changes have been merged. It's working fine now. Please have a try.
0.6.2 have been unblocked as well. 

I will close this issue and feel free to reopen if you still have problems.

**Comment by zimmertr** [comment]
This is happening again on a fresh v1.15.7 Kubernetes cluster deployed using `kubeadm`. 

```
$> export PATH=$PATH:/tmp/kubeflow/
export KF_NAME=test
export BASE_DIR=/opt/
export KF_DIR=${BASE_DIR}/${KF_NAME}
export CONFIG_URI="https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_k8s_istio.0.7.1.yaml"
mkdir -p ${KF_DIR}
cd ${KF_DIR}
```

```
$> sudo kfctl apply -V -f ${CONFIG_URI}
INFO[0000] Downloading https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_k8s_istio.0.7.1.yaml to /tmp/272735206/tmp.yaml  filename="utils/k8utils.go:169"
INFO[0000] Downloading https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_k8s_istio.0.7.1.yaml to /tmp/227049992/tmp_app.yaml  filename="configconverters/converters.go:71"
INFO[0000] App directory /Users/tj/tmp already exists    filename="coordinator/coordinator.go:274"
INFO[0000] Writing KfDef to kfctl_k8s_istio.0.7.1.yaml   filename="coordinator/coordinator.go:277"
INFO[0000] No name specified in KfDef.Metadata.Name; defaulting to tmp based on location of config file: /Users/tj/tmp/kfctl_k8s_istio.0.7.1.yaml.  filename="coordinator/coordinator.go:202"
INFO[0000]
****************************************************************
Notice anonymous usage reporting enabled using spartakus
To disable it
If you have already deployed it run the following commands:
  cd $(pwd)
  kubectl -n ${K8S_NAMESPACE} delete deploy -l app=spartakus

For more info: https://www.kubeflow.org/docs/other-guides/usage-reporting/
****************************************************************
  filename="coordinator/coordinator.go:120"
INFO[0000] Creating directory /Users/tj/tmp/.cache       filename="kfconfig/types.go:437"
INFO[0000] Fetching https://github.com/kubeflow/manifests/archive/v0.7-branch.tar.gz to /Users/tj/tmp/.cache/manifests  filename="kfconfig/types.go:485"
Error: failed to build kfApp from URI https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_k8s_istio.0.7.1.yaml: couldn't generate KfApp:  (kubeflow.error): Code 500 with message: could not sync cache. Error:  (kubeflow.error): Code 400 with message: couldn't download URI https://github.com/kubeflow/manifests/archive/v0.7-branch.tar.gz Error Error opening a gzip reader for /tmp/getter718595015/archive: EOF
Usage:
  kfctl apply -f ${CONFIG} [flags]

Flags:
  -f, --file string   Static config file to use. Can be either a local path:
                      		export CONFIG=./kfctl_gcp_iap.yaml
                      	or a URL:
                      		export CONFIG=https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_gcp_iap.0.7.0.yaml
                      		export CONFIG=https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_existing_arrikto.0.7.0.yaml
                      		export CONFIG=https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_aws.0.7.0.yaml
                      		export CONFIG=https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_k8s_istio.0.7.0.yaml
                      	kfctl apply -V --file=${CONFIG}
  -h, --help          help for apply
  -V, --verbose       verbose output default is false

failed to build kfApp from URI https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_k8s_istio.0.7.1.yaml: couldn't generate KfApp:  (kubeflow.error): Code 500 with message: could not sync cache. Error:  (kubeflow.error): Code 400 with message: couldn't download URI https://github.com/kubeflow/manifests/archive/v0.7-branch.tar.gz Error Error opening a gzip reader for /tmp/getter718595015/archive: EOF
```

**Comment by zimmertr** [comment]
https://github.com/kubeflow/kubeflow/issues/4642

**Comment by sitakantachaudhury** [comment]
Hi ,

what would be the solution for the above error.
